src/Service/Configurations/RuntimeConfigProvider.cs

```diff
@@ -228,6 +228,49 @@ public virtual bool TryLoadRuntimeConfigValue()
             return runtimeConfigJson;
         }
 
+        /// <summary>
+        /// Initialize the runtime configuration provider with the specified configurations.
+        /// This initialization method is used when the configuration is sent to the ConfigurationController
+        /// in the form of a string instead of reading the configuration from a configuration file.
+        /// This method assumes the connection string is provided as part of the configuration.
+        /// </summary>
+        /// <param name="configuration">The engine configuration.</param>
+        /// <param name="schema">The GraphQL Schema. Can be left null for SQL configurations.</param>
+        /// <param name="accessToken">The string representation of a managed identity access token</param>
+        /// <returns>true if the initialization succeeded, false otherwise.</returns>
+        public async Task<bool> Initialize(
+            string configuration,
+            string? schema,
+            string? accessToken)
+        {
+            if (string.IsNullOrEmpty(configuration))
+            {
+                throw new ArgumentException($"'{nameof(configuration)}' cannot be null or empty.", nameof(configuration));
+            }
+
+            if (RuntimeConfig.TryGetDeserializedRuntimeConfig(
```

Comment:
As this method assumes connection string is part of config, would it make sense to validate its existence here and throw if not provided?